packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -120,8 +171,23 @@ function composeMarks<TNodeChange>(
                     // TODO: preserve the insertions as muted
                     return 0;
                 }
+                case "MoveOut":
+                case "ReturnFrom":
+                    // The insert has been moved by `newMark`.
+                    // We can represent net effect of the two marks as an insert at the move destination.
+                    replaceMoveDest(
+                        moveEffects,
+                        getUniqueMoveId(newMark, newMark.revision ?? newRev, genId, moveEffects),
```

Comment:
`newMark.revision ?? newRev` is used 11 times in this function. Computing it once after `if (isSkipMark(newMark)) { ... }` would help tidy up.